### ISSUE TITLE
Which HTML spec to reference?

### ISSUE THREAD

**Issue by mavgit** [w3c-cg/webmediaapi#138]
WMAS2017 references [W3C HTML 5.1 2nd edition, REC, 3 October 2017](https://www.w3.org/TR/html51/).

WMAS2018 could reference:
a. [W3C HTML 5.1 2nd edition, REC, 3 October 2017](https://www.w3.org/TR/html51/) (No change.)
b. [W3C HTML 5.2, REC, 14 December 2017](https://www.w3.org/TR/html52/)
c. [W3C HTML 5.3, WD, 09 August 2018](https://www.w3.org/TR/html53/)
d. [WHATWG HTML Living Standard](url)

1. **HTML 5.3:** Since HTML 5.3 is only in Working Draft and there have been concerns about some 5.3 features, I believe that HTML 5.3 is not the best choice. However, I'd love for browser vendors and W3C folks to chime in.

2. **HTML 5.1 vs. HTML 5.2:** Considering only between W3C HTML 5.1 and W3C HTML 5.2, I understand that our spec should reference 5.2 because it is reasonably well supported by all 4 browsers. Again, I'd love for browser vendors and W3C folks to chime in.

3. **W3C HTML 5.2 vs. WHATWG HTML Living Standard:** So the real choice may boil down to referencing W3C 5.2 or switching to referencing WHATWG HTML Living Standard going forward.

**Comment by mavgit** [comment]
In the September 5, 2018 HATF call, there was some stated support for referencing HTML 5.2, some arguments against referencing the WHATWG HTML spec at this time and no support for any other options than HTML 5.2.

So, I'm calling for consensus for referencing HTML 5.2, but I'd like to hear from more folks.

**Comment by squapp** [comment]
note that it will be difficult to reference a version of WHATWG HTML spec, since it's a "living standard": 
https://github.com/whatwg/html/blob/master/FAQ.md#why-are-there-no-stable-snapshots-or-versions-of-the-standard

**Comment by mavgit** [comment]
@squapp Agree that is the case today, as stated in the WHATWG FAQ. I think there is a possibility that there will be a way to reference a dated version of the spec, depending on the outcome of W3C/WHATWG negotiations. Until that time, I recommend that we move to reference the W3C versions of HTML 5.2, as in the Call for Consensus.

This also closes out an action item from the 2018-09-18 WAVE face-to-face meeting in Amsterdam:

> The HATF chairs to determine whether it is possible to make a dated reference to the WHATWG HTML Living Standard.

It is not possible at this time.

**Comment by mavgit** [comment]
At the 2018-09-18 WAVE face-to-face meeting in Amsterdam, I believe that there was consensus to not reference the WHATWG HTML spec unless there is a way to make a dated reference.

**Comment by mavgit** [comment]
On the 2018-10-03 WAVE call, there was consensus to apply this pull request and there have been no dissenting comments on the issue since the call for consensus was posted two weeks earlier, so in accordance with our practice, I'm applying the pull request and closing the issue.

**Comment by mavgit** [comment]
https://github.com/w3c/webmediaapi/issues/138#issuecomment-422735589
> At the 2018-09-18 WAVE face-to-face meeting in Amsterdam, I believe that there was consensus to not reference the WHATWG HTML spec unless there is a way to make a dated reference."

I have learned that [WHATWG is now creating "review draft" snapshots as part of their IP practice](https://blog.whatwg.org/first-review-drafts). This could allow us to switch HTML and DOM to WHATWG Living Standards, but also make a non-spec-reference link to the latest snapshots for limited purposes, like IP and the testing contract.

I propose the form would be the Living Standard document reference in [brackets] followed by a parenthetical link to the latest review draft:
>HTML [[HTML](https://html.spec.whatwg.org)] ([review draft](https://html.spec.whatwg.org/review-drafts/2018-07/))
>DOM [[DOM](https://dom.spec.whatwg.org)] ([review draft](https://dom.spec.whatwg.org/review-drafts/2018-06/))

Note: Only the Living Standard would appear in the References section.

**Comment by mavgit** [comment]
Another thing we need to address in changing to the WHATWG HTML, involves specs that are chapters of the WHATWG HTML spec that were published as separate specs by W3C.

1. In some cases, we now have references to older W3C specs that are outdated versions of updated specs in sections of the WHATWG HTML spec. The following outdated, duplicate references would be deleted:
- Web Storage: Delete old [2016 W3C spec](https://www.w3.org/TR/webstorage/) which duplicates [WHATWG HTML section 11](https://html.spec.whatwg.org/multipage/webstorage.html#webstorage).
- Web Workers: Delete old [2015 W3C spec](https://www.w3.org/TR/workers/) which duplicates [WHATWG HTML section 10](https://html.spec.whatwg.org/multipage/workers.html#workers).
- Server-Sent Events: Delete old [2015 W3C spec](https://www.w3.org/TR/eventsource/) which duplicates [WHATWG HTML section 9.2](https://html.spec.whatwg.org/multipage/server-sent-events.html#server-sent-events).

2. In some cases, we now have references to individual chapters of the WHATWG HTML spec that would now be included in the single WHATWG HTML reference. The following duplicate references would be deleted:
- Web Sockets: Delete duplicate reference of [WHATWG HTML section 9.3](https://html.spec.whatwg.org/multipage/web-sockets.html#network).
- Cross-document messaging: Delete duplicate reference of [WHATWG HTML section 9.4](https://html.spec.whatwg.org/multipage/web-messaging.html#web-messaging).
- Channel messaging: Delete duplicate reference of [WHATWG HTML section 9.4](https://html.spec.whatwg.org/multipage/web-messaging.html#channel-messaging).

3. In one case, a section of the WHATWG HTML spec isn't implemented across all four code bases. The following would be listed as an exception under the WHATWG HTML reference:
- Server-Sent Events.

In summary, after switching to WHATWG HTML reference, we delete six references and add one exception.

**Comment by louaybassbouss** [comment]
Thanks @mavgit for you efforts just one comment from our side (Fraunhofer FOKUS) regarding the Living Standard reference and Review Draft Reference: the addition of the review draft link doesn’t help the testing work (https://github.com/cta-wave/WMAS2017). It is very difficult to select WPT tests for specific API review draft version. We prefer to  just include the reference to the WHATWG HTML Living Standard.

**Comment by andyburras** [comment]
I got the impression from the call that the Review Draft Reference was required to allow for supply contracts, i.e. a customer engaging a supplier needs to be able to state a definitive requirement in the contract.

**Comment by mavgit** [comment]
@andyburras 
>I got the impression from the call that the Review Draft Reference was required to allow for supply contracts

That was where we started, but towards the end of the discussion, I think people were realizing that the supply contracts face the same issue as the issue @louaybassbouss mentioned in https://github.com/w3c/webmediaapi/issues/138#issuecomment-433849076 regarding the test contract: since there's no way to select for tests based on review draft, we will inevitably include tests based on spec changes after the review draft. Therefore, if someone were to only implement to the review draft, it won't pass the tests.

I think the better approach that someone suggested on the call is that the supply contracts would reference our test suite and state that the implementation needs to pass the test suite. The supply contract could also state that the code should be based on a branch of the client platform code that is on or after the date of the spec publication.

In the end, the only fixed reference we have is that on the date of spec publication, we will (a) take a fixed snapshot of all the WPT tests on that date and (b) take a snapshot of each of the four browser releases on that date.

Since the WHATWG review drafts are from an earlier date than our publication and test snapshots, I don't think we should include them. I think we should just include the WHATWG spec reference without the review draft like this:

>HTML [[HTML](https://html.spec.whatwg.org)]
>DOM [[DOM](https://dom.spec.whatwg.org)]

**Comment by mavgit** [comment]
I call for consensus on pull request #181 which:
1. Updates HTML & DOM to WHATWG references, as described in https://github.com/w3c/webmediaapi/issues/138#issuecomment-433971427.
2. Deletes Web Storage, Web Workers, Server-Sent Events, Web Sockets, Cross-document messaging and Channel messaging references, since they're all in WHATWG HTML, as described in https://github.com/w3c/webmediaapi/issues/138#issuecomment-433652327.

Note: This doesn't include any review draft references, for reasons discussed in https://github.com/w3c/webmediaapi/issues/138#issuecomment-433849076 and https://github.com/w3c/webmediaapi/issues/138#issuecomment-433971427.

**Comment by poolec** [comment]
One concern I would have is that someone comparing the 2017 and 2018 WMAS documents might be puzzled that Web Storage, Web Workers etc. appear to have been removed.  For the items removed from section 3, I think it would be worth leaving a note to explain, at least for the 2018 revision.

For example, at the end of 3.10, add something like:

NOTE: Web Storage, Web Workers, Cross-document messaging and Channel messaging are also required as part of the HTML specification [HTML].

and at the end of 3.8, something like:

NOTE: Web Sockets is also required as part of the HTML specification [HTML].

**Comment by mavgit** [comment]
@poolec 
>For the items removed from section 3, I think it would be worth leaving a note to explain, at least for the 2018 revision.

Agree. I added two notes as you described in this commit: https://github.com/w3c/webmediaapi/pull/181/commits/e4d9856207a8ff1f1f7b6fdbf7a108bdb60e7b7b

**Comment by mavgit** [comment]
On the 2018-10-31 HATF call,  there was a suggestion to add a note to WMAS2018 that the versions of code bases will be published on the front page of the test suite.

**Comment by mavgit** [comment]
On the 2018-10-31 HATF call, we reviewed pull request #181 in detail and there was consensus to support applying #181 with the addition of the note on code base versions in https://github.com/w3c/webmediaapi/issues/138#issuecomment-434738708.

**Comment by mavgit** [comment]
Applied changes. Closing issue.

**Comment by poolec** [comment]
The two notes added to mention the specs that are now part of HTML need a minor fix to have proper references to the HTML spec (currently they just have text that reads [HTML] that isn't a reference).
